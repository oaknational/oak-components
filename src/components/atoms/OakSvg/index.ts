export { OakSvg } from "./OakSvg";
export type { OakSvgProps, OakSvgNames } from "./OakSvg";
