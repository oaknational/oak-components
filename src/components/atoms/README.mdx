# Atoms

## Rules for components in this folder

- Can import other atoms
- Imports from molecules and organisms are forbidden!
- Generic
- Unstyled
- Avoid embedded logic and state
- No sub-components
  (eg. Box)
