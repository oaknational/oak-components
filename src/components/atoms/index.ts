export * from "./OakBox";
export * from "./OakFlex";
export * from "./OakSpan";
export * from "./OakForm";
export * from "./OakImage";
export * from "./OakTypography";
export * from "./OakHeading";
export * from "./OakP";
export * from "./OakUL";
export * from "./OakOL";
export * from "./OakLI";
export * from "./OakLabel";
export * from "./OakIcon";
export * from "./OakGrid";
export * from "./OakGridArea";
export * from "./OakThemeProvider";
export * from "./OakMaxWidth";
export * from "./OakCloudinaryImage";
export * from "./OakKbd";
export * from "./OakGlobalStyle";
export * from "./OakAnchorTarget";
export * from "./OakFieldSet";
