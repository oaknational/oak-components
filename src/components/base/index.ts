export * from "./OakBox";
export * from "./OakFlex";
export * from "./OakBaseCard";
export * from "./OakSpan";
export * from "./OakForm";
export * from "./OakImage";
export * from "./OakTypography";
export * from "./OakHeading";
export * from "./OakP";
export * from "./OakUL";
export * from "./OakOL";
export * from "./OakLI";
export * from "./OakLabel";
