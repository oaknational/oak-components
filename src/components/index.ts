export * from "./base";
// export * from "./ui"; // uncomment this line once there are components in this folder
// export * from "./integrated"; // uncomment this line once there are components in this folder
