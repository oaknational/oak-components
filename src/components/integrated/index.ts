export * from "./OakQuizCheckBox";
export * from "./OakQuizTextInput";
export * from "./OakQuizRadioButton";
export * from "./OakQuizHint";
export * from "./OakQuizBottomNav";
export * from "./OakLessonTopNav";
export * from "./OakQuizFeedback";
export * from "./OakQuizCounter";
