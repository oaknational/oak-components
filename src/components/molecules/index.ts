export * from "./OakFieldError";
export * from "./OakLoadingSpinner";
export * from "./OakPrimaryButton";
export * from "./OakPrimaryInvertedButton";
export * from "./OakSecondaryButton";
export * from "./OakTertiaryButton";
export * from "./OakRadioButton";
export * from "./OakRadioGroup";
export * from "./OakCheckBox";
export * from "./OakTextInput";
export * from "./OakRoundIcon";
export * from "./OakTooltip";
export * from "./OakBackLink";
export * from "./OakBulletList";
export * from "./OakHandDrawnCard";
export * from "./OakHandDrawnCardWithIcon";
export * from "./OakHandDrawnHR";
export * from "./OakPromoTag";
export * from "./OakLessonInfoCard";
export * from "./OakCollapsibleContent";
export * from "./OakLink";
export * from "./OakSecondaryLink";
export * from "./OakDragAndDropInstructions";
export * from "./OakDraggable";
export * from "./OakDroppable";
export * from "./OakDraggableFeedback";
export * from "./OakOutlineAccordion";
export * from "./OakAccordion";
export * from "./OakModal";
export * from "./OakModalCenter";
export * from "./OakCardWithHandDrawnBorder";
export * from "./OakSecondaryButtonAsRadio";
export * from "./OakButtonAsRadioGroup";
