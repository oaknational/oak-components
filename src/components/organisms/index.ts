export * from "./pupil";
export * from "./teacher";
export * from "./OakSubjectIcon";
export * from "./OakPrimaryNavItem";
export * from "./OakPrimaryNav";
export * from "./OakCookieBanner";
export * from "./OakCookieSettingsModal";
export * from "./OakCookieConsentProvider";
export * from "./OakCookieConsent";
