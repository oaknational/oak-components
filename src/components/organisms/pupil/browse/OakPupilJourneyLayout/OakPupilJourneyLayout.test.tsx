import React from "react";
import "@testing-library/jest-dom";
import { create } from "react-test-renderer";

import {
  OakPupilJourneyLayout,
  getBackgroundUrlForSection,
  PupilJourneySectionName,
  Phase,
} from "./OakPupilJourneyLayout";

import renderWithTheme from "@/test-helpers/renderWithTheme";
import { OakThemeProvider } from "@/components/atoms";
import { oakDefaultTheme } from "@/styles";

describe("OakPupilJourneyLayout", () => {
  it("should render successfully", () => {
    const { getByText } = renderWithTheme(
      <OakPupilJourneyLayout sectionName={"tier-listing"}>
        <p>Hello World</p>
      </OakPupilJourneyLayout>,
    );
    expect(getByText("Hello World")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const tree = create(
      <OakThemeProvider theme={oakDefaultTheme}>
        <OakPupilJourneyLayout sectionName={"tier-listing"}>
          <p>Hello World</p>
        </OakPupilJourneyLayout>
      </OakThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("getBackgroundUrlForSection", () => {
  it("should return the correct background image for lesson-listing primary", () => {
    expect(getBackgroundUrlForSection("lesson-listing", "primary")).toBe(
      "https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1734522868/pupil-journey/confetti-pink.svg",
    );
  });

  it("should return the correct background image for lesson-listing secondary", () => {
    expect(getBackgroundUrlForSection("lesson-listing", "secondary")).toBe(
      "https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1734519662/pupil-journey/confetti-lavender.svg",
    );
  });

  it("should return the correct background image for unit-listing primary", () => {
    expect(getBackgroundUrlForSection("unit-listing", "primary")).toBe(
      "https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1712912389/pupil-journey/line-pink.svg",
    );
  });

  it("should return the correct background image for unit-listing secondary", () => {
    expect(getBackgroundUrlForSection("unit-listing", "secondary")).toBe(
      "https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1712912348/pupil-journey/line-lavender.svg",
    );
  });

  it("should return the correct background image for subject-listing", () => {
    expect(getBackgroundUrlForSection("subject-listing")).toBe(
      "https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1715356384/pupil-journey/line-mint.svg",
    );
  });

  it("should return the correct background image for year-listing", () => {
    expect(getBackgroundUrlForSection("year-listing")).toBe(
      "https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1734522739/pupil-journey/confetti-mint.svg",
    );
  });
});

describe("OakPupilJourneyLayout backgroundColor", () => {
  const renderComponent = (
    sectionName: PupilJourneySectionName,
    phase?: Phase,
  ) =>
    renderWithTheme(
      <OakPupilJourneyLayout sectionName={sectionName} phase={phase}>
        <p>Test Content</p>
      </OakPupilJourneyLayout>,
    );

  it("should return bg-decorative4-very-subdued for lesson-listing primary", () => {
    const { container } = renderComponent("lesson-listing", "primary");
    expect(container.firstChild).toHaveStyle("background: rgb(245, 233, 242)");
  });

  it("should return bg-decorative3-very-subdued for lesson-listing secondary", () => {
    const { container } = renderComponent("lesson-listing", "secondary");
    expect(container.firstChild).toHaveStyle("background: rgb(227, 233, 251)");
  });

  it("should return bg-decorative4-very-subdued for unit-listing primary", () => {
    const { container } = renderComponent("unit-listing", "primary");
    expect(container.firstChild).toHaveStyle("background: rgb(245, 233, 242)");
  });

  it("should return bg-decorative3-very-subdued for unit-listing secondary", () => {
    const { container } = renderComponent("unit-listing", "secondary");
    expect(container.firstChild).toHaveStyle("background: rgb(227, 233, 251)");
  });

  it("should return bg-decorative4-very-subdued for tier-listing primary", () => {
    const { container } = renderComponent("tier-listing", "primary");
    expect(container.firstChild).toHaveStyle("background: rgb(245, 233, 242)");
  });

  it("should return bg-decorative3-very-subdued for tier-listing secondary", () => {
    const { container } = renderComponent("tier-listing", "secondary");
    expect(container.firstChild).toHaveStyle("background: rgb(227, 233, 251)");
  });

  it("should return bg-decorative1-main for subject-listing", () => {
    const { container } = renderComponent("subject-listing");
    expect(container.firstChild).toHaveStyle("background: rgb(190, 242, 189)");
  });

  it("should return bg-decorative1-main for year-listing", () => {
    const { container } = renderComponent("year-listing");
    expect(container.firstChild).toHaveStyle("background: rgb(190, 242, 189)");
  });
});
