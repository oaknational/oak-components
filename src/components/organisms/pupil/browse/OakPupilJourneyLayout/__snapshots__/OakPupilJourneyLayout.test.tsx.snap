// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakPupilJourneyLayout matches snapshot 1`] = `
.c0 {
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:750px) {
  .c0 {
    padding-left: 1.5rem;
  }
}

@media (min-width:750px) {
  .c0 {
    padding-right: 1.5rem;
  }
}

@media (min-width:1280px) {
  .c2 {
    background-image: url();
    background-repeat: no-repeat;
    background-position-x: center;
    background-size: 100%;
  }
}

<div>
  <div
    class="c0 c1 c2"
  >
    <p>
      Hello World
    </p>
  </div>
</div>
`;
